src/CommunityToolkit.Maui/Views/Expander/Expander.shared.cs

```diff
@@ -0,0 +1,221 @@
+
+	/// <inheritdoc />
+	public IView? Header
+	{
+		get => (IView?)GetValue(HeaderProperty);
+		set => SetValue(HeaderProperty, value);
+	}
+
+	/// <inheritdoc />
+	public new IView? Content
+	{
+		get => (IView?)GetValue(ContentProperty);
+		set => SetValue(ContentProperty, value);
+	}
+
+	/// <inheritdoc />
+	public bool IsExpanded
+	{
+		get => (bool)GetValue(IsExpandedProperty);
+		set => SetValue(IsExpandedProperty, value);
+	}
+
+	/// <inheritdoc />
+	public ExpandDirection Direction
+	{
+		get => (ExpandDirection)GetValue(DirectionProperty);
+		set => SetValue(DirectionProperty, value);
+	}
+
+	/// <summary>
+	/// Command parameter passed to the <see cref="Command"/>
+	/// </summary>
+	public object? CommandParameter
+	{
+		get => GetValue(CommandParameterProperty);
+		set => SetValue(CommandParameterProperty, value);
+	}
+
+	/// <summary>
+	/// Command to execute when <see cref="IsExpanded"/> changed.
+	/// </summary>
+	public ICommand? Command
+	{
+		get => (ICommand?)GetValue(CommandProperty);
+		set => SetValue(CommandProperty, value);
+	}
+
+	static void OnHeaderPropertyChanged(BindableObject bindable, object oldValue, object newValue)
+		=> ((Expander)bindable).Configure();
+
+	static void OnContentPropertyChanged(BindableObject bindable, object oldValue, object newValue)
+		=> ((Expander)bindable).Configure();
+
+	static void OnIsExpandedPropertyChanged(BindableObject bindable, object oldValue, object newValue)
+		=> ((Expander)bindable).Configure();
+
+	static void OnDirectionPropertyChanged(BindableObject bindable, object oldValue, object newValue)
+		=> ((Expander)bindable).Configure();
+
+	void IExpander.ExpandedChanged(bool isExpanded)
+	{
+		if (Command is not null && Command.CanExecute(CommandParameter))
+		{
+			Command.Execute(CommandParameter);
+		}
+
+		tappedEventManager.HandleEvent(this, new Core.ExpandedChangedEventArgs(isExpanded), nameof(ExpandedChanged));
+	}
+
+	void Configure()
+	{
+		if (Header is null || Content is null)
+		{
+			return;
+		}
+
+		SetGestures();
+		Layout();
+		UpdateSize();
+	}
+
+	void Layout()
+	{
+		var grid = new Grid();
+		grid.Children.Add(Header);
+		switch (Direction)
+		{
+			case ExpandDirection.Down:
+				grid.RowDefinitions.Add(new RowDefinition(GridLength.Auto));
+				grid.SetRow(Header, 0);
+				if (IsExpanded)
+				{
+					grid.Children.Add(Content);
+					grid.RowDefinitions.Add(new RowDefinition(GridLength.Auto));
+					grid.SetRow(Content, 1);
+				}
+				break;
+			case ExpandDirection.Up:
+				if (IsExpanded)
+				{
+					grid.Children.Add(Content);
+					grid.RowDefinitions.Add(new RowDefinition(GridLength.Auto));
+					grid.SetRow(Content, 0);
+					grid.RowDefinitions.Add(new RowDefinition(GridLength.Auto));
+					grid.SetRow(Header, 1);
+				}
+				else
+				{
+					grid.RowDefinitions.Add(new RowDefinition());
+					grid.SetRow(Header, 0);
+				}
+				break;
+		}
+
+		base.Content = grid;
```

Comment:
I'v tried this code on Angroid and do have exception here:

```
The specified child already has a parent. You must call removeView() on the child's parent first
```

To avoid this code deeded before setting the `Content`:
```cs
#if ANDROID
		var cnv = base.Content?.Handler?.PlatformView as Android.Views.ViewGroup;
		cnv?.RemoveAllViews();
#endif
```